src/main/java/database/DataInstancesBuilder.java

```diff
@@ -0,0 +1,119 @@
+package database;
+
+import utils.RandomString;
+import weka.core.Attribute;
+import weka.core.FastVector;
+import weka.core.Instance;
+import weka.core.Instances;
+
+import javax.xml.crypto.Data;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.*;
+
+/**
+ * Created by Timotej on 31-Oct-15.
+ *
+ * This class builds up weka dataset (Instances)
+ */
+public class DataInstancesBuilder {
+
+    private int batchID = -1;
+    private int subjectID = -1;
+    private Map<Integer, String> features = new HashMap<Integer, String>();
+    private RandomString rndStr = new RandomString((int) Math.ceil(Math.pow(
+            features.size(),
+            1 / RandomString.alphabetSize())) + 1);
+
+    /**
+    REQUIRED tells which batch to use
+     */
+    public DataInstancesBuilder forBatch(int batchID) {
+        this.batchID = batchID;
+        return this;
+    }
+
+    /**
+    OPTIONAL, if used it just gets data for that subject
+     */
+    public DataInstancesBuilder forSubject(int subjectID) {
+        this.subjectID = subjectID;
+        return this;
+    }
+
+    /**
+     REQUIRED tells which features to put into the dataset
+     Call multiple times to add more features
+     */
+    public DataInstancesBuilder withFeature(int featureID) {
```

Comment:
I've gone with the other approach. Enums are just a nightmare, and also using names gives a more fail soft approach. We want that as it will be more useful in training and it isn't production code anyway.